### PR TITLE
ci: renovate GH workflows; JDK 17

### PR DESCRIPTION
**Pull request by ennru**
- Current versions of GH actions.
- Use coursier setup-action
- Add JDK17 to the matrices.

Notice: release runs on JDK 8

Aligns with
- https://github.com/akka/akka/pull/31730
- https://github.com/akka/akka/pull/31734

### CODE REVIEW

**Comment by ennru** [comment]
Branch protection must be updated when merging this.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -28,13 +28,13 @@ jobs:
...
-          java-version: adopt@1.8.0-275
-
       - name: Cache Coursier cache
-        uses: coursier/cache-action@v5
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Set up JDK 8
+        uses: coursier/setup-action@v1.3.0
+        with:
+          jvm: adopt:1.8.0-275
```
Comment (patriknw) [inline .github/workflows/release.yml:37]: not changing to temurin for the release?

Comment (ennru) [reply .github/workflows/release.yml:37]: I kept it, for now. I didn't see any `-release 8` flags, otherwise JDK 11 is our new default for releasing.

Comment (patriknw) [reply .github/workflows/release.yml:37]: ok, then changing that would be a different task

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**